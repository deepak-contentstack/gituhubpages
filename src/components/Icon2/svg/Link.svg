<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M16.6666 7.33337L7.33325 16.6667" stroke="#475161"/>
<path d="M10 6.66671L13.7978 2.86891C15.7504 0.916285 18.9162 0.916286 20.8689 2.86891L21.1311 3.13117C23.0838 5.0838 23.0838 8.24962 21.1311 10.2022L17.3333 14" stroke="#475161"/>
<path d="M13.3333 18L9.53545 21.7978C7.58283 23.7505 4.41701 23.7505 2.46438 21.7978L2.20212 21.5356C0.249497 19.583 0.249498 16.4171 2.20212 14.4645L5.99992 10.6667" stroke="#475161"/>
</svg>
