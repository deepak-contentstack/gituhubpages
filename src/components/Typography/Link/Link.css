@import '../../../../tokens/variables.scss';

.Link{
    display: inline-block;
    font-size: $size-font-large;
    &:hover{
        &>a{
            color: $color-font-link;
        }
        .Link__dashed,.Link__solid {
            border-bottom: 1px solid rgba(108,92,231,.8);
      }
    }
    &.ExternalLink{
        &:hover{
            a{
                color: $color-font-link;
                >svg{
                    path{
                        fill: $color-font-link;
                    }
                }
            }
        }
    }
    a {
        text-decoration: none;
        .Link__dashed{
            border-bottom: 1px dashed rgba(108,92,231,.8);
        }
        .Link__solid{
            border-bottom: none;
        }
    }
    &--link {
        a {
            color: $color-font-link;
        }
            &:hover{
        a{
            color: #5D50BF;
        }
            }
    }
    &--white {
        a {
            color: $color-font-white;
        }
        &:hover {
            a {
                color: $color-font-white!important;
            }
        }
    }
    &--black {
        a {
            color: $color-font-black;
        }
    }
    &--base {
        a {
            color: $color-font-base;
        }
    }
    &--tiny{
        font-size: $size-font-tiny;
    }

    &--small{
        font-size: $size-font-small;
    }

    &--medium{
        font-size: $size-font-medium;
    }

    &--large{
        font-size: $size-font-large;
    }

    &--bold{
        font-weight: $font-weight-bold;
    }

    &--semi-bold{
        font-weight: $font-weight-semi-bold;
    }

    &--regular{
        font-weight: $font-weight-regular;
    }
}

