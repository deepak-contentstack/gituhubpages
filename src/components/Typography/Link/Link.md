# Link Component

## Best Practices

- Use fontSize and fontWeight properties to highlight linked text.
