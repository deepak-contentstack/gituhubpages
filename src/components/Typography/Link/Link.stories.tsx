import React from 'react'
import Link from '../Link/Link'
import notes from './Link.md'
import { action } from '@storybook/addon-actions';
import DocSubData from '../../../utils/components/DocSubData';

const fontColorList: any = ['black', 'link', 'base', 'white'];
const fontSizeList: any = ['tiny', 'small', 'medium', 'large'];
const fontWeightList: any = ['bold', 'semi-bold', 'regular'];
const underLineTypeList: any = ['dashed', 'solid'];
const targetList: any = ['__blank', '_self', '_parent', '_top'];
const typeList: any = ['external'];
const componentDescription = 'Links are navigational elements that let you traverse to web-based destinations or assets.'
const usageDescription = 'Link allows you to customize anchor elements. You can use them on their own or inline with text.'

export default {
  title: 'Components / Typography/Link',
  component: Link,
  argTypes: {
    href: {
      control: 'text'
    },
    fontSize: {
      control: {
        type: "select",
        options: fontSizeList
      }
    },
    fontColor: {
      control: {
        type: "select",
        options: fontColorList
      }
    },
    fontWeight: {
      control: {
        type: "select",
        options: fontWeightList
      }
    },
    underline: {
      control: 'boolean'
    },
    underLineType: {
      control: {
        type: "select",
        options: underLineTypeList
      }
    },
    target: {
      control: {
        type: "select",
        options: targetList
      }
    },
    type: {
      control: {
        type: "select",
        options: typeList
      }
    },
    className: {
      control: "disabled"
    },
    label: {
      control: 'text'
    },
    style: {
      control: "disabled"
    },
    children: {
      control: "disabled"
    },
    cbOnClick: {
      control: "disabled"
    },
    iconName: {
      control: "disabled"
    }
  },
  parameters: {
    componentSubtitle: <DocSubData componentDescription={componentDescription} usageDescription={usageDescription} />,
    design: {
      type: 'figma',
      url: 'https://www.figma.com/file/NUaPd7QxttcJAQfx9KYjBu/01-%E2%80%A2-Atoms-%26-Properties?node-id=1263%3A9'
    },
    notes: { notes }
  }
}

const tableDisable = { table: { disable: true } }

export const Default = (args) => {

  return (
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit{' '}
      <Link
        href={args.href}
        cbOnClick={action('cbOnClick triggered')}
        underLineType={args.underLineType}
        underline={args.underline}
        fontColor={args.fontColor}
        fontSize={args.fontSize}
        fontWeight={args.fontWeight}
        iconName={args.iconName}
        className={args.className}
        target={args.target}>
        {args.label}
      </Link>
      {' '}sed do eiusmod tempor incididunt ut labore et dolore magna.
    </p>
  )
}
Default.args = {
  underline: 'true',
  label: "Learn more"
}
Default.argTypes = {
  children: tableDisable,
  testId: tableDisable,
}

export const ExternalLink = (args) => {
  return (
    <Link href={args.href} type={args.type} underline={args.underline} underLineType={args.underLineTypeList}>Learn more</Link>
  )
}
ExternalLink.args = {
  type: "external",
  underline: 'true',
}
ExternalLink.argTypes = {
  children: tableDisable,
  cbOnClick: tableDisable,
  target: tableDisable,
  fontWeight: tableDisable,
  fontSize: tableDisable,
  fontColor: tableDisable,
  style: tableDisable,
  className: tableDisable,
  testId: tableDisable,
}
ExternalLink.parameters = {
  docs: {
    storyDescription: 'You can use this component to display an external link with an icon. And, you need to specify the URL you want to open when the component is clicked on.',
  },
}

export const onClick = (args) => (
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit <Link cbOnClick={action("cbOnClick triggered")} href={null} underLineType={args.underLineType} underline={args.underline} >Learn more</Link> sed do eiusmod tempor incididunt ut labore et dolore magna.
  </p>
)
onClick.args = {
  underline: 'true',
}
onClick.argTypes = {
  children: tableDisable,
  target: tableDisable,
  type: tableDisable,
  fontWeight: tableDisable,
  fontSize: tableDisable,
  fontColor: tableDisable,
  iconName: tableDisable,
  style: tableDisable,
  className: tableDisable,
  testId: tableDisable,
}
onClick.parameters = {
  docs: {
    storyDescription: 'You can use this component in scenarios where you need a function to be triggered when the link/text will be clicked on.',
  },
}