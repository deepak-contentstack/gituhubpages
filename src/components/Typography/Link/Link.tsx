import React, { CSSProperties, ReactNode } from 'react'
// import Icon from '../../Icon2';
import cn from 'classnames'
import './Link.css'

export type LinkProps = {
  children: ReactNode,
  /**
   * Specify the URL where the text is to be linked.
   */
  href: string,
  /**
   *  Pass the function to be triggered when the link is clicked on.
   */
  cbOnClick?: Function,
  /**
   * Set the type of link to “external” when you want the icon and text to be made clickable
   */
  type?: 'external',
  /**
   * Specify where to display the end response.
   */
  target?: "_blank" | "_self" | "_parent" | "_top",
  /**
   * Make the link text bolder.
   */
  fontWeight?: "bold" | "regular" | "semi-bold",
  /**
   * Set the font-size of link text
   */
  fontSize?: "tiny" | "small" | "medium" | "large"
  /**
   * Add styles to the component.
   */
  style?: CSSProperties,
  /**
   * Add a custom icon, set this prop to the name of the icon.
   */
  iconName?: string,
  /**
   *  Pass the class names to be appended to this prop.
   */
  className?: string,
  /**
   * Display an underline below the hyperlink, set this prop to “true”, else set it to “false”.
   */
  underline?: boolean,
  /**
   * Change the underline style of the link to either “dashed” or “solid”.
   */
  underLineType?: 'dashed' | 'solid'
  /**
   * Change the color of the link text.
   */
  fontColor: "white" | "base" | "link" | "black"
  /**
   * Pass an ID that you can use for testing purposes.
   */
  testId?: string;
}

const Link = (props: LinkProps) => {
  const { children, className, href, type, fontSize, underline, iconName, underLineType, fontWeight, fontColor, cbOnClick, testId, ...rest } = props;
  const isExternal = type === 'external';
  const target = props.target ? props.target : "_blank"

  const classNames = cn('Link', className, {
    'ExternalLink': isExternal,
    [`Link--${fontWeight}`]: fontWeight,
    [`Link--${fontSize}`]: fontSize,
    [`Link--${fontColor}`]: fontColor,
  })

  const handleOnClick = () => {
    if (cbOnClick) {
      cbOnClick();
    }
  }
  return (
    <div className={classNames} onClick={handleOnClick} data-test-id={testId} {...rest}>
      <a href={href} target={target} className="flex-v-center">
        <span className={`${underline ? `Link__${underLineType}` : ""}`}>
          {children}
        </span>
        {/* {isExternal &&
          // <Icon className="ml-5" icon={iconName} />
        } */}
      </a>
    </div>
  )
}

Link.defaultProps = {
  iconName: "NewTab",
  underLineType: "dashed",
  target: "_blank",
  underline: false,
  fontWeight: "regular",
  fontColor: "base",
  fontSize: "large",
  testId: 'cs-link'
} as Partial<LinkProps>;

export default Link;