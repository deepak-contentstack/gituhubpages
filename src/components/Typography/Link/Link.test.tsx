import React from 'react'
import renderer from 'react-test-renderer'
import Link from './Link'
import { mount } from 'enzyme'
describe('Link', () => {
  it('render default Link component', () => {
    const wrapper = renderer.create(<Link href="/">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('render External Link component', () => {
    const wrapper = renderer.create(<Link href="/" type="external">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('render Fontweight regular', () => {
    const element = mount(<Link href="/" fontWeight="regular">Learn More</Link>);
    expect(element.find('div').props().className.includes('regular')).toBe(true);
  });
  it('render Fontweight bold', () => {
    const element = mount(<Link href="/" fontWeight="bold">Learn More</Link>);
    expect(element.find('div').props().className.includes('bold')).toBe(true);
  });
  it('render Fontweight semi-bold', () => {
    const element = mount(<Link href="/" fontWeight="semi-bold">Learn More</Link>);
    expect(element.find('div').props().className.includes('semi-bold')).toBe(true);
  });

  it('render Fontsize medium', () => {
    const element = mount(<Link href="/" fontSize="medium">Learn More</Link>);
    expect(element.find('div').props().className.includes('medium')).toBe(true);
  });

  it('render Fontsize tiny', () => {
    const element = mount(<Link href="/" fontSize="tiny">Learn More</Link>);
    expect(element.find('div').props().className.includes('tiny')).toBe(true);
  });

  it('render Fontsize small', () => {
    const element = mount(<Link href="/" fontSize="small">Learn More</Link>);
    expect(element.find('div').props().className.includes('small')).toBe(true);
  });

  it('render Fontsize large', () => {
    const element = mount(<Link href="/" fontSize="large">Learn More</Link>);
    expect(element.find('div').props().className.includes('large')).toBe(true);
  });

  it('render Fontcolor link', () => {
    const element = mount(<Link href="/" fontColor="link">Learn More</Link>);
    expect(element.find('div').props().className.includes('link')).toBe(true);
  });

  it('render Fontcolor base', () => {
    const element = mount(<Link href="/" fontColor="base">Learn More</Link>);
    expect(element.find('div').props().className.includes('base')).toBe(true);
  });

  it('render Fontcolor white', () => {
    const element = mount(<Link href="/" fontColor="white">Learn More</Link>);
    expect(element.find('div').props().className.includes('white')).toBe(true);
  });

  it('render Fontcolor black', () => {
    const element = mount(<Link href="/" fontColor="black">Learn More</Link>);
    expect(element.find('div').props().className.includes('black')).toBe(true);
  });

  it('render target _blank', () => {
    const wrapper = renderer.create(<Link href="/" target="_blank">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('render target _self', () => {
    const wrapper = renderer.create(<Link href="/" target="_self">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('render target _parent', () => {
    const wrapper = renderer.create(<Link href="/" target="_parent">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('render target _top', () => {
    const wrapper = renderer.create(<Link href="/" target="_top">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('render underline true', () => {
    const wrapper = renderer.create(<Link href="/" underline={true}>Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it('render underline false', () => {
    const wrapper = renderer.create(<Link href="/" underline={false}>Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('render underlinetype dashed', () => {
    const element = mount(<Link href="/" underline={true} underLineType="dashed">Learn More</Link>);
    expect(element.find('span').props().className.includes('dashed')).toBe(true);
  });
  it('render underlinetype solid', () => {
    const element = mount(<Link href="/" underline={true} underLineType="solid">Learn More</Link>);
    expect(element.find('span').props().className.includes('solid')).toBe(true);
  });

  it('render iconName', () => {
    const wrapper = renderer.create(<Link href="/" type="external" iconName="NewTab">Learn More</Link>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('It should trigger the cbOnClick of link', () => {
    const mockFunction = jest.fn();
    const LinkOnClick = ({ onClickAction }) => {
      return (
        <Link href="/" cbOnClick={onClickAction}>Learn More</Link>
      )
    };
    const element = mount(<LinkOnClick onClickAction={mockFunction} />);
    element.find('div').simulate('click');
    expect(mockFunction).toHaveBeenCalledTimes(1);
  })

  it('renders class name', () => {
    const DefaultLink = () => {
      return (
        <Link className="Link">Learn more</Link>
      )
    }
    const tree = renderer.create(<DefaultLink />).toJSON();
    expect(tree.props.className.includes("Link")).toBe(true);
  })

  it('render additional style', () => {
    const DefaultLink = () => {
      return (
        <Link style={{ color: 'red' }}>Learn more</Link>
      )
    }
    const tree = renderer.create(<DefaultLink />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});