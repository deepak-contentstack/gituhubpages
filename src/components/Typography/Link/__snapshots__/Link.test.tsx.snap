// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link render External Link component 1`] = `
<div
  className="Link ExternalLink Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_blank"
>
  <a
    className="flex-v-center"
    href="/"
    target="_blank"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render additional style 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
  target="_blank"
>
  <a
    className="flex-v-center"
    target="_blank"
  >
    <span
      className=""
    >
      Learn more
    </span>
  </a>
</div>
`;

exports[`Link render default Link component 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_blank"
>
  <a
    className="flex-v-center"
    href="/"
    target="_blank"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render iconName 1`] = `
<div
  className="Link ExternalLink Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_blank"
>
  <a
    className="flex-v-center"
    href="/"
    target="_blank"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render target _blank 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_blank"
>
  <a
    className="flex-v-center"
    href="/"
    target="_blank"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render target _parent 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_parent"
>
  <a
    className="flex-v-center"
    href="/"
    target="_parent"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render target _self 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_self"
>
  <a
    className="flex-v-center"
    href="/"
    target="_self"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render target _top 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_top"
>
  <a
    className="flex-v-center"
    href="/"
    target="_top"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render underline false 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_blank"
>
  <a
    className="flex-v-center"
    href="/"
    target="_blank"
  >
    <span
      className=""
    >
      Learn More
    </span>
  </a>
</div>
`;

exports[`Link render underline true 1`] = `
<div
  className="Link Link--regular Link--large Link--base"
  data-test-id="cs-link"
  onClick={[Function]}
  target="_blank"
>
  <a
    className="flex-v-center"
    href="/"
    target="_blank"
  >
    <span
      className="Link__dashed"
    >
      Learn More
    </span>
  </a>
</div>
`;
